### PR TITLE
Dump of initial implementation of Recommendation Manager

### PR DESCRIPTION
**Pull request by ShishirH**
Signed-off-by: Shishir Halaharvi <shalahar@redhat.com>

### CODE REVIEW

**Comment by ShishirH** [comment]
YAML and code changes to add hpo_algo_impl and step

File: manifests/autotune-config-crd.yaml [hunk 1]
```diff
@@ -102,6 +102,9 @@ spec:
                 value_type:
                   description: 'Value type of the query response (double, string, int)'
                   type: string
+                step:
+                  description: 'Step of the tunable'
```
[the remark this answers is not in the document]
Comment (ShishirH) [reply manifests/autotune-config-crd.yaml:106]: In the code, yes. Right now there are no default values. Yes, we will need to update all the autotune-config yamls and add steps

File: src/main/java/com/autotune/dependencyAnalyzer/k8sObjects/SlaInfo.java [hunk 2]
```diff
@@ -59,13 +61,15 @@ public SlaInfo(String slaClass,
 			this.direction = direction;
 		else throw new InvalidValueException("Invalid direction for autotune kind");
 
+		this.hpoAlgoImpl = hpoMlImpl;
```
Comment (dinogun) [inline src/main/java/com/autotune/dependencyAnalyzer/k8sObjects/SlaInfo.java:64]: Same for the Algo, if nothing is set, then we have to set `optuna_tpe`.

**Review by dinogun — CHANGES_REQUESTED** [review]

**Comment by dinogun** [comment]
This PR is now no longer required

**Pull request closed without merging**